ConfiguredBehaviourTest

**com.hazelcast.client.connectionstrategy.ConfiguredBehaviourTest.testReconnectModeONSingleMember**
```java
com.hazelcast.client.HazelcastClientNotActiveException: Partition does not have an owner. partitionId: 31
	at com.hazelcast.client.spi.impl.ClientInvocation.notifyException(ClientInvocation.java:186)
	at com.hazelcast.client.spi.impl.ClientInvocation.invokeOnSelection(ClientInvocation.java:153)
	at com.hazelcast.client.spi.impl.ClientInvocation.retry(ClientInvocation.java:171)
	at com.hazelcast.client.spi.impl.ClientInvocation.run(ClientInvocation.java:159)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:439)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:98)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:206)
	at com.hazelcast.util.executor.LoggingScheduledExecutor$LoggingDelegatingFuture.run(LoggingScheduledExecutor.java:128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:895)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:918)
	at java.lang.Thread.run(Thread.java:662)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
	at ------ submitted from ------.(Unknown Source)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.resolveAndThrowIfException(ClientInvocationFuture.java:96)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.resolveAndThrowIfException(ClientInvocationFuture.java:33)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:155)
	at com.hazelcast.client.spi.ClientProxy.invokeOnPartition(ClientProxy.java:188)
	at com.hazelcast.client.spi.ClientProxy.invoke(ClientProxy.java:182)
	at com.hazelcast.client.proxy.ClientMapProxy.putInternal(ClientMapProxy.java:516)
	at com.hazelcast.client.proxy.ClientMapProxy.put(ClientMapProxy.java:508)
	at com.hazelcast.client.proxy.ClientMapProxy.put(ClientMapProxy.java:311)
	at com.hazelcast.client.connectionstrategy.ConfiguredBehaviourTest.testReconnectModeONSingleMember(ConfiguredBehaviourTest.java:157)
Caused by: java.io.IOException: Partition does not have an owner. partitionId: 31
	at com.hazelcast.client.spi.impl.ClientSmartInvocationServiceImpl.invokeOnPartitionOwner(ClientSmartInvocationServiceImpl.java:42)
	at com.hazelcast.client.spi.impl.ClientInvocation.invokeOnSelection(ClientInvocation.java:143)
	at com.hazelcast.client.spi.impl.ClientInvocation.retry(ClientInvocation.java:171)
	at com.hazelcast.client.spi.impl.ClientInvocation.run(ClientInvocation.java:159)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:439)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:98)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:206)
	at com.hazelcast.util.executor.LoggingScheduledExecutor$LoggingDelegatingFuture.run(LoggingScheduledExecutor.java:128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:895)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:918)
	at java.lang.Thread.run(Thread.java:662)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
```
https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/9592/testReport/junit/com.hazelcast.client.connectionstrategy/ConfiguredBehaviourTest/testReconnectModeONSingleMember/

```
java.lang.AssertionError: CountDownLatch failed to complete within 120 seconds, count left: 1
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:1009)
	at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:1002)
	at com.hazelcast.test.HazelcastTestSupport.assertOpenEventually(HazelcastTestSupport.java:994)
	at com.hazelcast.client.connectionstrategy.ConfiguredBehaviourTest.testReconnectModeASYNCSingleMemberStartLate(ConfiguredBehaviourTest.java:200)
```

https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/10669/testReport/junit/com.hazelcast.client.connectionstrategy/ConfiguredBehaviourTest/testReconnectModeASYNCSingleMemberStartLate/

--- a/hazelcast-client/src/test/java/com/hazelcast/client/connectionstrategy/ConfiguredBehaviourTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/connectionstrategy/ConfiguredBehaviourTest.java
@@ -179,12 +179,9 @@ public class ConfiguredBehaviourTest extends ClientTestSupport {
         HazelcastInstance hazelcastInstance = hazelcastFactory.newHazelcastInstance();
 
         ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getNetworkConfig().setConnectionAttemptLimit(Integer.MAX_VALUE);
         clientConfig.getConnectionStrategyConfig().setReconnectMode(ASYNC);
         HazelcastInstance client = hazelcastFactory.newHazelcastClient(clientConfig);
-
-        assertTrue(client.getLifecycleService().isRunning());
-
-        hazelcastInstance.shutdown();
 
         client.getLifecycleService().addLifecycleListener(new LifecycleListener() {
             @Override
@@ -195,8 +192,11 @@ public class ConfiguredBehaviourTest extends ClientTestSupport {
             }
         });
 
+        hazelcastInstance.shutdown();
+
         hazelcastFactory.newHazelcastInstance();
 
+        assertTrue(client.getLifecycleService().isRunning());
         assertOpenEventually(reconnectedLatch);
 
         client.getMap(randomMapName());

